Library/Homebrew/cmd/config.rb

```diff
@@ -119,9 +119,15 @@ def describe_system_ruby
     s
   end
 
-  def java_version
-    java = `java -version 2>&1`.lines.first.chomp
-    java =~ /java version "(.+?)"/ ? $1 : java
+  def describe_java
+    if which("java").nil?
+      "N/A"
+    elsif `/usr/libexec/java_home --failfast 2>&1` =~ /Unable to find any JVMs/ && !$?.success?
```

Comment:
From my tests actually you can rely just on the exit code rather than needing the match.